<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>461</width>
    <height>540</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Обработка цифровых изображений</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="rBtnNonLinearFilter">
        <property name="text">
         <string>Нелинейный фильр</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rBtnNiblackThresholding">
        <property name="text">
         <string>Локальная пороговая обработка (Ниблак)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rBtnAdaptiveThresholding">
        <property name="text">
         <string>Адаптивная пороговая обработка</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="labelArea">
          <property name="text">
           <string>Окрестность</string>
          </property>
          <property name="buddy">
           <cstring>sBoxArea</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sBoxArea">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="gBoxNonLinearFilter">
        <property name="title">
         <string>Нелинейный фильтр</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QRadioButton" name="rBtnMedian">
           <property name="text">
            <string>Медианный</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rBtnMinimum">
           <property name="text">
            <string>Максимума</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rBtnMaximum">
           <property name="text">
            <string>Минимума</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gBoxNiblackThresholding">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Локальная пороговая обработка (Ниблак)</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="labelK">
           <property name="text">
            <string>k</string>
           </property>
           <property name="buddy">
            <cstring>sBoxK</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="sBoxK">
           <property name="minimum">
            <double>-99.000000000000000</double>
           </property>
           <property name="value">
            <double>-0.200000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gBoxAdaptiveThresholding">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Адаптивная пороговая обработка</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="labelA">
           <property name="text">
            <string>a</string>
           </property>
           <property name="buddy">
            <cstring>sBoxA</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="sBoxA">
           <property name="minimum">
            <double>-9.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9.000000000000000</double>
           </property>
           <property name="value">
            <double>0.300000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnProcess">
        <property name="text">
         <string>Обработать</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnSelectPicture">
        <property name="text">
         <string>Выбрать изображение</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="labelPicture">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="labelResult">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>rBtnNonLinearFilter</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNonLinearFilter</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>418</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnNonLinearFilter</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNiblackThresholding</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnNonLinearFilter</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxAdaptiveThresholding</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnNiblackThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNonLinearFilter</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnNiblackThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNiblackThresholding</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnNiblackThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxAdaptiveThresholding</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>339</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnAdaptiveThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNonLinearFilter</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>305</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnAdaptiveThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxNiblackThresholding</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>283</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>283</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rBtnAdaptiveThresholding</sender>
   <signal>clicked(bool)</signal>
   <receiver>gBoxAdaptiveThresholding</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
